pkg/controller/gittrack/git_creds_test.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 Pusher Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package gittrack
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	farosv1alpha1 "github.com/pusher/faros/pkg/apis/faros/v1alpha1"
+)
+
+var _ = Describe("GitTrack Suite", func() {
+	Describe("createRepoRefFromCreds", func() {
```

Comment:
Tests in general look very good, one thing that came to my mind is that we aren't checking extra fields aren't set, could we maybe construct the expected `repo` and compare the entire struct rather than fields within it? WDYT? This allows us to ensure the fields that should be empty, are empty 😄

Reply:
💪 Better expectations FTW!